qiskit/algorithms/time_evolution/evolution_base.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for quantum time evolution."""
+
+from abc import ABC, abstractmethod
+
+from qiskit.algorithms.time_evolution.problems.evolution_problem import EvolutionProblem
+from qiskit.algorithms.time_evolution.evolution_result import EvolutionResult
+
+
+class EvolutionBase(ABC):
+    """Base class for quantum time evolution."""
+
+    @abstractmethod
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
```

Comment:
This is just a remark, and I have no clear opinion. I was wondering if it makes more sense to have something like (let's imagine a `QTE` class inherits from this `EvolutionBase`):
``` 
qte = QTE(evolution_problem)
evol_result = qte.evolve()
```
Instead of the current:
``` 
qte = QTE()
evol_result = qte.evolve(evolution_problem)
```
This would make it closer to, for example, the way VQE and other algorithms are implemented, where an instance of the `VQE` class is "assigned" to a specific problem when initialized, if that makes sense. I am not super familiar with the problems that this framework is going to tackle, but I after looking at other Qiskit solvers, I believe this is the way it's usually done (maybe there isn't a clear argument for it either).

Reply:
Another possible idea would be to extract the `time` attribute from the `EvolutionProblem` class, and have something like:

```
qte = QTE(evolution_problem)
evol_result = qte.evolve(time)
```

Reply:
Yeah, so far we have been leaning towards solving different problems with the same instance of a QTE algorithm. But if analogous algorithms in Terra do it differently then we might reconsider. Thanks for pointing this out.

Reply:
I think that 
```qte = QTE(evolution_problem)
evol_result = qte.evolve()```
would be a good way to streamline the use of QTE algorithms.
I'm also open to extracting time to the `evolve` function.

Reply:
I checked that it is not unified within Terra - look at `AmplitudeAmplifier` or `PhaseEstimator`. They do not accept a problem or any problem-related data in their constructors.
Given that, I would keep our approach here as is.